store/postgres/src/connection_pool.rs

```diff
@@ -19,10 +28,18 @@ pub fn create_connection_pool(
     postgres_url: String,
     pool_size: u32,
     logger: &Logger,
+    registry: Arc<dyn MetricsRegistry>,
```

Comment:
I'm a bit rusty on setting up metrics counters; can we just pass the counter into `create_connection_pool()` rather than the entire `MetricsRegistry`?

Reply:
I'd rather not create the counter in `main` and keep the counter specific knowledge in the crate/components that needs it

Reply:
Yep, sounds good. 
Looking closer now and seeing the pattern of passing in the registry is consistent with the other counters too. 👍